packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/mira/MiraController.java

```diff
@@ -262,21 +315,14 @@ public ResponseEntity<List<DKG>> search(
 			@RequestParam("q") final String q,
 			@RequestParam(required = false, name = "limit", defaultValue = "10") final Integer limit,
 			@RequestParam(required = false, name = "offset", defaultValue = "0") final Integer offset) {
+		final ResponseEntity<List<DKG>> response;
 		try {
-			final ResponseEntity<List<DKG>> response = proxy.search(q, limit, offset);
-			if (response.getStatusCode().is2xxSuccessful()) {
-				return ResponseEntity.ok(response.getBody());
-			}
-			return ResponseEntity.internalServerError().build();
+			response = proxy.search(q, limit, offset);
 		} catch (final FeignException e) {
-			final String error = "An error occurred searching DKGs";
-			final int status = e.status() >= 400 ? e.status() : 500;
-			log.error(error, e);
-			throw new ResponseStatusException(org.springframework.http.HttpStatus.valueOf(status), error);
-		} catch (final Exception e) {
-			log.error("Unable to fetch DKG", e);
-			return ResponseEntity.internalServerError().build();
+			throw handleMiraFeignException(e, "concepts", "query", q, "mira.concepts.bad-query");
 		}
+
+		return new ResponseEntity(response.getBody(), response.getStatusCode());
```

Comment:
Is there any risk here of returning a 500 or something with the body? I suppose that wouldnt matter?

Reply:
a 500 should be thrown as a feignException. If we get to this point, we have a 2xx response. I didn't want to convert to a 200 and lose that.